### PR TITLE
tests/raftstore: add a partition function for Cluster

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

File: tests/raftstore/cluster.rs [hunk 1]
```diff
@@ -458,6 +459,43 @@ impl<T: Simulator> Cluster<T> {
         assert!(status_resp.has_region_detail());
         status_resp.take_region_detail()
     }
+
+    // NOTE: if you have set transport hooks before, call this function will overwrite them
+    pub fn partition(&mut self, s1: Arc<HashSet<u64>>, s2: Arc<HashSet<u64>>) {
+        for node_id in s1.as_ref() {
```
Comment (siddontang) [inline tests/raftstore/cluster.rs]: in &s1

Comment (tiancaiamao) [inline tests/raftstore/cluster.rs]: compiler will complain

        error: the trait bound `&std::sync::Arc<std::collections::HashSet<u64>>: std::iter::Iterator` is not satisfied [E0277]

Comment (BusyJay) [inline tests/raftstore/cluster.rs]: What about `iter` ?

Comment (tiancaiamao) [inline tests/raftstore/cluster.rs]: iter() is ok

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**